test/integration/puppetlabs/services/legacy_routes/legacy_routes_test.clj

```diff
@@ -31,20 +33,57 @@
 
 (deftest ^:integration legacy-routes
   (testing "The legacy web routing service properly handles old routes."
-    (bootstrap/with-puppetserver-running app
-      {}
-      (is (= 200 (:status (http-get "/v2.0/environments"))))
-      (is (= 200 (:status (http-get "/production/node/localhost"))))
-      (is (= 200 (:status (http-get "/production/certificate_statuses/all")))))))
+    (bootstrap/with-puppetserver-running-with-mock-jruby-puppet-fn
+     "JRuby mocking is safe ,here because we're only interested in validating
+     that the request is routed from a legacy to a new endpoint and the
+     translation is expected to occur at the Clojure layer, not in Ruby."
+     app
+     {}
+     (partial jruby-testutils/create-mock-jruby-puppet
+              (fn [request]
+                (JRubyPuppetResponse.
+                 (Integer. 200)
+                 (str "request routed to: " (get request "uri"))
+                 "text/plain"
+                 "1.2.3")))
```

Comment:
This function (or something very similar) is repeated several places.  Would it be worthwhile to extract it into a helper?